JOSESwift/Sources/PBES2KeyEncryptionMode.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  PBES2KeyEncryptionMode.swift
+//  JOSESwift
+//
+//  Created by Tobias Hagemann on 08.12.23.
+//
+//  ---------------------------------------------------------------------------
+//  Copyright 2023 Airside Mobile Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  ---------------------------------------------------------------------------
+//
+
+import Foundation
+
+struct PBES2KeyEncryptionMode {
+    typealias KeyType = String
```

Comment:
From [RFC-7518](https://datatracker.ietf.org/doc/html/rfc7518#section-4.8):

> The PBES2 password input is an
   octet sequence; if the password to be used is represented as a text
   string rather than an octet sequence, the UTF-8 encoding of the text
   string MUST be used as the octet sequence.

I guess we had to choose to either have `String` or `Data` as a data type here. I'd love to hear the reasoning behind choosing `String` in this case. Not saying one is better than the other, just genuinely interested in the reasoning.